docs/proposals/project-scoped-appset.md

```diff
@@ -0,0 +1,114 @@
+In this use case, ApplicationSet is the perfect match for us because they can deploy an appset for each microservice and automatically they are deployed into all the clusters with the required configuration (also enabling disaster recovery scenarios in other providers/regions thanks to the generators)
+
+As a platform team, we provide to the teams the tooling like ArgoCD, Harbor, observability,... This means that we manage an ArgoCD installation shared for all the teams, managing their clusters from the same place (based on RBAC). We'd like to allow the usage of appsets for the teams, but due to the global scope of them, teams can deploy things into other teams clusters because appsets ignores any kind of limitation.
+
+### Goals
+
+Add support for deploying ApplicationSets inside a AppProject and limit the interaction with elements inside it. 
+This includes also the RBAC verification
+
+### Non-Goals
+
+This doesn't include any change in the UI/cli to make them user friendly
+
+## Proposal
+
+The proposal for achieving this goal is to add a "required" `project` field (read more below) into the ApplicationSet and based on it, limit the access to the resources during the Applications generation. 
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-addons
+  namespace: argocd
+spec:
+  project: sample-project # new field
+  generators:
+  - ...
+```
+
+As we want to maintain the option of deploying global ApplicationSet (without any restriction), administrators will be able to deploy appsets without setting the project field. For achieving this, we will check if the user has access to **ALL** `applicationsets` across **ALL** the projects (*/*) and only that case, they will be able to deploy `applicationsets` without setting the project:
+
+```
+p, role:admin, applicationsets, create, */*, allow
+p, role:admin, applicationsets, update, */*, allow
+```
+
+Other option is to be explicit and create an specific permission for this:
+
+```
+p, role:admin, applicationsets, unscoped, *, allow
+```
+
+> Note: `unscoped` is a proposal, but maybe `global` or any other name is better.
+
+Inside the appset controller, the controller will check if appset has set the `project` field and it can produce 2 different behaviors:
+
+1. `project` isn't set: In this case, nothing will change from current behavior
+2. `project` is set: In this case, **ALL** the resources involved in the Application generation have to be scoped to the project where the appset is deployed. 
+In case of cluster/git generators the cluster/repos have to be project scoped, but in other generators, the controller will check if the resource is allowed for the project (for example, using a list generator the controller will block forbidden clusters/repos)
```

Comment:
> in other generators, the controller will check if the resource is allowed for the project (for example, using a list generator the controller will block forbidden clusters/repos)

Isn't the Application controller already enforcing that? Or do you envision the value of `spec.project` differing from the value of `spec.template.spec.project`?

Reply:
If I have 2 AppProject with one cluster per project (Project A with Cluster A and Project B with Cluster B), currently I can generate (and sync if `permitOnlyProjectScopedClusters != true`) and application for Project B in Cluster A.
That's the triggering reason for my feature request xD
My idea is to limit this possibility during the generation process instead of blocking the sync after the application is generated. I mean, the application for Project B never should be generated in Cluster A

Reply:
You are right, if I enable `permitOnlyProjectScopedClusters` in the project, I can block the app creation in other projects' clusters, but I still have the problem of the repositories

Reply:
Agree with @crenshaw-dev . I think applicationset just need to make sure all generated applications belongs to appset project. Application controller will enforce everything else.